src/main/java/xades4j/production/SignerBES.java

```diff
@@ -244,14 +248,27 @@ public final XadesSignatureResult sign(
             // with its value set to: http://uri.etsi.org/01903#SignedProperties."
 
             String digestAlgUri = algorithmsProvider.getDigestAlgorithmForDataObjsReferences();
-            if (null == digestAlgUri)
+            if (StringUtils.isNullOrEmptyString(digestAlgUri))
             {
                 throw new NullPointerException("Digest algorithm URI not provided");
             }
 
+            // Use same canonicalization URI as specified in the ds:CanonicalizationMethod for Signature.
+            String canonAlg = this.algorithmsProvider.getCanonicalizationAlgorithmForSignature().getUri();
+
             try
             {
-                signature.addDocument('#' + signedPropsId, null, digestAlgUri, null, QualifyingProperty.SIGNED_PROPS_TYPE_URI);
+                Transforms transforms = null;
+
+                if (!StringUtils.isNullOrEmptyString(canonAlg))

```

Comment:
Hi Luis,

Thanks for the feedback! Based on your suggestion, I will remove the extra checks and always add the canonicalization algorithm. Please verify the commit https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762 . Thanks!